health-services/ExerciseSampleCompose/app/src/main/java/com/example/exercisesamplecompose/data/HealthServicesRepository.kt

```diff
@@ -71,7 +76,15 @@ class HealthServicesRepository @Inject constructor(
         }
     }
 
-    fun startExercise() = serviceCall { startExercise() }
+    fun startExercise() = serviceCall {
+        try {
+            startExercise()
+            errorState.value = null
+        } catch (e: Exception) {
+            errorState.value = e.message
+            e.printStackTrace()
```

Comment:
remove or log properly?

Reply:
Done, added a log but also kept the stacktrace.